app/main/views/briefs.py

```diff
@@ -174,19 +174,24 @@ def update_brief_status(brief_id, action):
     brief = Brief.query.filter(
         Brief.id == brief_id
     ).first_or_404()
-
-    if brief.framework.status != 'live':
-        abort(400, "Framework is not live")
+    framework_status = brief.framework.status
 
     action_to_status = {
         'publish': 'live',
         'withdraw': 'withdrawn',
         'cancel': 'cancelled',
         'unsuccessful': 'unsuccessful'
     }
-    if brief.status != action_to_status[action]:
-        previousStatus = brief.status
-        brief.status = action_to_status[action]
+    new_status = action_to_status[action]
+
+    if new_status in ['live', 'withdrawn'] and framework_status != 'live':
+        abort(400, "Framework is not live")
+    if new_status in ['cancelled', 'unsuccessful'] and framework_status not in ['live', 'expired']:
+        abort(400, "Framework is not live or expired")
```

Comment:
Nice catch, I hadn't thought about the need for cancelling briefs on older frameworks. We should check that the award path allows this to?

Reply:
(Just checked - the award path is fine)